src/shipping/views/PriceRatesUpdate/PriceRatesUpdate.tsx

```diff
@@ -344,45 +361,32 @@ export const PriceRatesUpdate: React.FC<PriceRatesUpdateProps> = ({
             />
           </Button>
         }
-        onZipCodeAssign={() => openModal("add-range")}
-        onZipCodeUnassign={id =>
-          openModal("remove-range", {
-            id
-          })
-        }
+        onPostalCodeInclusionChange={onPostalCodeInclusionChange}
+        onPostalCodeAssign={() => openModal("add-range")}
+        onPostalCodeUnassign={code => {
+          if (code.id !== undefined) {
+            setCodesToDelete([...codesToDelete, code.id]);
+            rate.postalCodeRules = rate.postalCodeRules.filter(
+              rule => rule.id !== code.id
+            );
+          } else {
+            rate.postalCodeRules = rate.postalCodeRules.filter(
+              rule => rule.start !== code.start && rule.end !== code.end
```

Comment:
Seen this used a few times in some components, should be a utils function like 
```suggestion
     const getRulesUnmatchingPostalCode = (codeToCompare) => 
        (obj: { start, end }) => obj.start !== codeToCompare.start 
        && rule.end !== code.ToCompareEnd
```

And then do 
```suggestion
          filter(getRulesUnmatchingPostalCode(code))
```